javaparser-core/src/main/java/com/github/javaparser/printer/PrettyPrinterConfiguration.java

```diff
@@ -31,26 +33,49 @@
  */
 public class PrettyPrinterConfiguration {
     public static final int DEFAULT_MAX_ENUM_CONSTANTS_TO_ALIGN_HORIZONTALLY = 5;
-    
+
     private boolean orderImports = false;
     private boolean printComments = true;
     private boolean printJavadoc = true;
     private boolean columnAlignParameters = false;
     private boolean columnAlignFirstMethodChain = false;
-    private String indent = "    ";
+    private int indentSize = 4;
     private String endOfLineCharacter = EOL;
     private Function<PrettyPrinterConfiguration, PrettyPrintVisitor> visitorFactory = PrettyPrintVisitor::new;
     private int maxEnumConstantsToAlignHorizontally = DEFAULT_MAX_ENUM_CONSTANTS_TO_ALIGN_HORIZONTALLY;
 
+    /**
+     * Set the string to use for indenting. For example: "\t", "    ", "".
+     * @deprecated use setIndentSize
+     */
+    @Deprecated
+    public PrettyPrinterConfiguration setIndent(String indent) {
+        throw new NotImplementedException();
```

Comment:
How about instead the following?
- warn about deprecation if you do deprecation warnings
- if argument is spaces only, set `indentSize` to the length
- if argument contains anything else, throw exception (probably most users have couple of spaces or tabs, latter will maybe be added by my pending PR if you accept it, everything else is probably OK if it is not supported anymore)

Would make exisiting code simply continue to work like before besides a deprecation warning.

Reply:
I'll get tabs working properly with my PR, yes, as soon as you merged this, I'll probably adapt to the latest changes and submit it for your review. :-)